M276 Combat Toolbelt Rig

Sulaboy opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

Adds the M276 pattern combat toolbelt rig as to the game as a new belt for combat engineers and some other roles.

The M276 pattern combat toolbelt is an alternate version of the M276 pattern toolbelt with some combat capabilities. This rig is capable of carrying fives tool and has a holster for a handgun. 

# Explain why it's good for the game

Provides engineers with more choices for tool storage.


# Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots & Videos</summary>

![combatbelt showcase](https://user-images.githubusercontent.com/75421103/218943761-a188b2b2-2125-45eb-a15e-5eb9c7b02c95.png)

The newest iteration of this belt has total slots, and cannot fit pistol magazines.
</details>


**Disclaimer, I barely know how to code. If I need to make any tweaks please be patient with me.**


# Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->
<!-- If you add a name after the ':cl', that name will be used in the changelog. You must add your CKEY after the CL if your GitHub name doesn't match. Be sure to properly mark your PRs to prevent unnecessary GBP loss. Maintainers freely reserve the right to remove and add tags should they deem it appropriate. -->

:cl:
add: Added item M276 pattern combat toolbelt rig as a point buy item (15 points) to combat engineer vendor.
imageadd: Sprites for M276 pattern combat toolbelt rig
/:cl:

<!-- Both :cl:'s are required for the changelog to work! -->

Zonespace27 commented on code/modules/projectiles/guns/pistols.dm:
```diff
@@ -47,7 +47,7 @@
 /obj/item/weapon/gun/pistol/m4a3
 	name = "\improper M4A3 service pistol"//1911
-	desc = "An M4A3 Service Pistol, once the standard issue sidearm of the Colonial Marines but has recently been replaced with the 88 Mod 4 combat pistol. Fires 9mm pistol rounds."
+	desc = "An M4A3 Service Pistol, a stinky test pistol for testing changing code with a running server and being stinky."
```
.

Sulaboy replied:
That wasn't supposed to be there.

Zonespace27 commented on code/modules/projectiles/guns/rifles.dm:
```diff
@@ -37,7 +37,7 @@
 /obj/item/weapon/gun/rifle/m41a
 	name = "\improper M41A pulse rifle MK2"
-	desc = "The standard issue rifle of the Colonial Marines. Commonly carried by most combat personnel. Uses 10x24mm caseless ammunition."
+	desc = "Stinky Gun Test Gun Test Stinky. Commonly carried by most combat personnel. Uses 10x24mm caseless ammunition."
```
.

Sulaboy replied:
Fixed

morrowwolf reviewed (changes requested):
GA review:

This is far too powerful and power creep in its current state. This belt needs to be balanced alongside other belts rather than be a straight upgrade.

Sulaboy commented:
> GA review:
> 
> This is far too powerful and power creep in its current state. This belt needs to be balanced alongside other belts rather 
than be a straight upgrade.

Didn't get this to you before you got off for the night. Here is the picture with the belt capacity comparisons. I'll definitely be cutting out a magazine and slot from the belt at least or both if you think that's the way to go. Let me know if you would like it to be a point buy as well. I'll make any changes to this on Friday.

![asdf](https://user-images.githubusercontent.com/75421103/218965935-ab3bfd5e-47f9-4ee3-8bfd-164e32a6924e.png)

Sulaboy commented:
Removed a slot from the belt to put it more in line with other belts, and added a point cost. Removed any addition of the belt from other vendors and WO spawns.

morrowwolf reviewed (commented):
GA Approved. Usual disclaimer: maintainer team can and will change this item in whatever way (including completely reverted) if it effects balance in a way unforeseen.

NewyearnewmeUwu commented:
IMO, this should be the lifesaver bag to the medical rig or vice versa. A sidegrade, not an upgrade or a downgrade. There are some things the medical rig legitimately does better, but most of the time the livesaver is the nobrainer pick.

Agameofscones reviewed (changes requested):
should be `+combatutility_half`. Current is incorrect as `+combatutility`. (sprite state names)

Otherwise sprites checkout. Ping me when you've fixed it and I'll approve.

Zonespace27 commented on code/game/objects/items/storage/belt.dm:
```diff
@@ -1546,6 +1546,50 @@
+	name = "\improper M276 pattern combat toolbelt rig"
+	desc = "The M276 pattern combat toolbelt rig is an alternative load-bearing equipment of the USCM for engineers conducting repairs within combat zones. It consists of a modular belt with various clips and pouches for tools along with a holster for a sidearm. Due to the bulk of the sidearm, it is unable to hold as many tools as its standard counterpart."
+	storage_slots = 6
+	icon_state = "combatutility"
```
```suggestion
	icon_state = "combatutility"
```

Zonespace27 commented:
Looks good
